.travis.yml

```diff
@@ -8,22 +9,43 @@ env:
   - BRANCH=0.19.6
   - BRANCH=0.20.2
   # This is the latest working Nim version against which Nimble is being tested
+  # Update in cache:directories as well
   - BRANCH=#2565d3d102efd21ba02ed1f3b96d892fe2637d2b
 
 cache:
   directories:
+    - "$HOME/.nimble/bin"
     - "$HOME/.choosenim/toolchains/nim-0.19.6"
     - "$HOME/.choosenim/toolchains/nim-0.20.2"
+    - "$HOME/.choosenim/toolchains/nim-#2565d3d102efd21ba02ed1f3b96d892fe2637d2b"
 
 install:
-  - export CHOOSENIM_CHOOSE_VERSION=$BRANCH
-  - |
-      curl https://nim-lang.org/choosenim/init.sh -sSf > init.sh
-      sh init.sh -y
-
-before_script:
   - export CHOOSENIM_NO_ANALYTICS=1
   - export PATH=$HOME/.nimble/bin:$PATH
+  - |
+    if ! type -P choosenim &> /dev/null; then
+      if [[ "$TRAVIS_OS_NAME" == "windows" ]]; then
+        # Latest choosenim binary doesn't have
+        #   https://github.com/dom96/choosenim/pull/117
+        #   https://github.com/dom96/choosenim/pull/135
+        #
+        # Using custom build with these PRs for Windows
+        curl -L -s "https://bintray.com/genotrance/binaries/download_file?file_path=choosenim.exe" -o choosenim.exe
+        curl -L -s "https://bintray.com/genotrance/binaries/download_file?file_path=libeay32.dll" -o libeay32.dll
+        curl -L -s "https://bintray.com/genotrance/binaries/download_file?file_path=ssleay32.dll" -o ssleay32.dll
+        ./choosenim.exe $BRANCH -y
+        cp ./choosenim.exe $HOME/.nimble/bin/.
+        # 0.19.6 and 0.20.2 have nimble with issue #665
+        # Updating nimble to avoid failure with tests for #280 and #524
+        cd `./choosenim.exe show | grep Path | cut -f 8 -d " "` && ./koch.exe --latest nimble && cd -
```

Comment:
Nice, very thorough. But it seems very wrong to me that our test suite depends on another separate Nimble that we have in the PATH. This CI tests Nimble after all... can we do something about this instead of having these workarounds?

Reply:
I've created https://github.com/nimble-test/issue280and524/pull/3 to fix this. Once you approve that, I can kick off the testing for this PR.

Reply:
PTAL - had to create https://github.com/nimble-test/issue280and524/pull/4 to do this right.